docs/book/ahead-of-time-factories.md

```diff
@@ -0,0 +1,106 @@
+# Ahead of Time Factories 
```

Comment:
We need versioned documentation, so this must be placed in a subdirectory: `docs/book/v4/ahead-of-time-factories.md`
Create the folder and add the file, I will do the rest.

Reply:
Done. I duplicated all files and created both `v3` and `v4` folders.

Reply:
Great, I will do the rest.